msal/application.py

```diff
@@ -701,6 +707,10 @@ def _validate_ssh_cert_input_data(self, data):
                     "you must include a string parameter named 'key_id' "
                     "which identifies the key in the 'req_cnf' argument.")
 
+    def _setup_before_first_network_call(self):
```

Comment:
Perhaps this can be renamed as `_get_client(self)` and also internally have the same "reentrance" trick we used in authority.py? That way the callers might be able to just do `return self._get_client().obtain_token_by_xyz(...)`.